packages/shared/src/get-deep-element-from-point.ts

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+export const getDeepElementFromPoint = (
+    x: number,
+    y: number
+): Element | null => {
+    let target = document.elementFromPoint(x, y);
+    while (target?.shadowRoot) {
```

Comment:
Would it be valuable to include an escape hatch via a maximum depth? Could even put the maximum depth at something like 100 but it seems like a nice protection.

Reply:
I’m not sure of a lot of cases where this is actually functionality that a developer actually means to be leveraging and without more than the one we currently support I don’t want to over optimize here.